pkg/curatedpackages/bundle_test.go

```diff
@@ -0,0 +1,168 @@
+		Spec: packagesv1.PackageBundleControllerSpec{
+			ActiveBundle: activeBundle,
+		},
+	}
+	packageBundle := packagesv1.PackageBundle{
+		Spec: packagesv1.PackageBundleSpec{
+			Packages: []packagesv1.BundlePackage{
+				{
+					Name: "harbor",
+				},
+			},
+		},
+	}
+
+	return &bundleTest{
+		WithT:         gomega.NewWithT(t),
+		ctx:           context.Background(),
+		kubectl:       k,
+		bundleManager: &bm,
+		Command: &Bundle{
+			KubeConfig:    kubeConfig,
+			KubeVersion:   kubeVersion,
+			Kubectl:       k,
+			BundleManager: &bm,
+			CliVersion:    version.Info{GitVersion: "v1.0.0"},
+			Registry:      registry,
+		},
+		bundleCtrl:    &bundleCtrl,
+		packageBundle: &packageBundle,
+		activeBundle:  activeBundle,
+		registry:      registry,
+	}
+}
+
+func TestGetLatestBundleFromClusterSucceeds(t *testing.T) {
+	tt := newBundleTest(t)
+	tt.kubectl.EXPECT().ApplyResources(tt.ctx, gomock.Any()).Return(convertJsonToBytes(tt.bundleCtrl), nil)
+	tt.kubectl.EXPECT().ApplyResources(tt.ctx, gomock.Any()).Return(convertJsonToBytes(tt.packageBundle), nil)
+
+	tt.Command.Source = Cluster
+	result, err := tt.Command.GetLatestBundle(tt.ctx)
+	tt.Expect(err).To(gomega.BeNil())
+	tt.Expect(result.Spec.Packages[0].Name).To(gomega.BeEquivalentTo(tt.packageBundle.Spec.Packages[0].Name))
+}
+
+func TestGetLatestBundleFromRegistrySucceeds(t *testing.T) {
+	tt := newBundleTest(t)
+	tt.registry.EXPECT().GetRegistryBaseRef(tt.ctx).Return("test_host/test_env/test_controller", nil)
+
+	tt.Command.Source = Registry
+	result, err := tt.Command.GetLatestBundle(tt.ctx)
+	tt.Expect(err).To(gomega.BeNil())
+	tt.Expect(result.Spec.Packages[0].Name).To(gomega.BeEquivalentTo(tt.packageBundle.Spec.Packages[0].Name))
+}
+
+func TestGetLatestBundleFromUnknownSourceFails(t *testing.T) {
+	tt := newBundleTest(t)
+	tt.Command.Source = "Unknown"
+	_, err := tt.Command.GetLatestBundle(tt.ctx)
+	tt.Expect(err).To(gomega.MatchError(gomega.ContainSubstring("unknown source")))
+}
+
+func TestLatestBundleFromClusterUnknownBundle(t *testing.T) {
+	tt := newBundleTest(t)
+	tt.kubectl.EXPECT().ApplyResources(tt.ctx, gomock.Any()).Return(convertJsonToBytes(tt.bundleCtrl), nil)
+	tt.kubectl.EXPECT().ApplyResources(tt.ctx, gomock.Any()).Return(bytes.Buffer{}, errors.New("error reading bundle"))
+	tt.Command.Source = Cluster
+	_, err := tt.Command.GetLatestBundle(tt.ctx)
+	tt.Expect(err).To(gomega.MatchError(gomega.ContainSubstring("error reading bundle")))
+}
+
+func convertJsonToBytes(obj interface{}) bytes.Buffer {
+	b, _ := json.Marshal(obj)
+	return *bytes.NewBuffer(b)
+}
+
+type mockBundleManager struct{}
+
+// generated via
+// impl -dir $GOPATH/pkg/mod/github.com/aws/eks-anywhere-packages@v0.1.1/pkg/bundle 'bm *mockBundleManager' bundle.Manager
+
+// IsActive returns true if the given namespace and name matche the active
+// bundle's.
+func (bm *mockBundleManager) IsActive(ctx context.Context, client client.Client, namespacedName types.NamespacedName) (bool, error) {
```

Comment:
could we have a different package for the mocks?
Any reason why we can't use `gomock` for this one?